content/post/2024-02-29-stepping-down.md

```diff
@@ -0,0 +1,58 @@
+- Eventually we added what one would expect every p2p library to have - support for hole punching.
+  This was a huge effort.
+  We rolled our own relay implementation, a custom coordination protocol, various tricks around address discovery and mapping, ...
+  Long story short, [we invented lots of new wheels](https://blog.ipfs.tech/2022-01-20-libp2p-hole-punching/) ...
+  Really, we should have started off of [ICE](https://en.wikipedia.org/wiki/Interactive_Connectivity_Establishment).
+- rust-libp2p was long missing a QUIC transport implementation.
+  [Eventually](https://github.com/libp2p/rust-libp2p/issues/2883) we were able to deliver this based on the quinn project, which I am very happy about, bringing lots of performance improvements to rust-libp2p users.
+- I managed the [specification and implementation of WebRTC in libp2p](https://github.com/libp2p/specs/issues/220). While a great match for a p2p library, unfortunately up until today, it has not seen larger adoption.
+- A tiny fraction of IPFS today is powered by [a rust-libp2p based IPFS bootstrap node](https://blog.ipfs.tech/2023-rust-libp2p-based-ipfs-bootstrap-node/) handling ~35k connections on a medium-sized machine.
+- A major portion of my time in 2023 went into an [automated performance setup for the many libp2p implementations](https://github.com/libp2p/test-plans/blob/master/perf/README.md).
+  The outcome in my eyes is both [visually appealing](https://observablehq.com/@libp2p-workspace/performance-dashboard) and impactful from a performance perspective (see final item).
+- We shipped many developer experience improvements, taming some of the complexity inherent to peer-to-peer networking e.g. the [`SwarmBuilder`](https://github.com/libp2p/rust-libp2p/pull/4120).
+- Security is hard, especially on the internet, doubly so when in a peer-to-peer setting.
+  A lot of our work went into security fixes, especially around DOS attacks.
+  See e.g. a [major memory exhaustion attack](https://github.com/libp2p/rust-libp2p/security/advisories/GHSA-jvgw-gccv-q5p8) and [a talk](https://max-inden.de/post/2022-11-02-dos-defense-dos-and-donts/) inspired by this work.
+- Talking about talks, I gave [quite a couple](https://max-inden.de/tags/talk/) on peer-to-peer networking and libp2p and helped organize related conferences, e.g. the FOSDEM networking devroom.
+- As a final big contribution, the before mentioned automated performance benchmark setup enabled me to land a major improvement to libp2p's bandwidth performance.
+  Implementing [stream receive window auto-tuning in the Rust Yamux implementation](https://github.com/libp2p/rust-yamux/pull/176) improved throughput on high bandwidth connections from 33 Mbit/s to 1.3 Gbit/s.
+
+During my time the rust-libp2p maintainer team went from 4 down to 1, up to 2, 3, back to 4, and now down to 1.
+I hope for another positive trend to follow.
```

Comment:
```suggestion
I hope for the positive trend to continue.
```

Reply:
Oh, but the current trend is not positive. I.e. I hope for the number of maintainers to increase again.